### PR TITLE
refine MPPTaskCancelFunctor

### PR DESCRIPTION
**Pull request by windtalker**
Signed-off-by: xufei <xufeixw@mail.ustc.edu.cn>

### What problem does this PR solve?

Issue Number: ref #5638

Problem Summary:
refine `MPPTaskCancelFunctor`
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/Mpp/MPPTaskManager.cpp [hunk 1]
```diff
@@ -70,29 +70,18 @@ MPPTaskPtr MPPTaskManager::findTaskWithTimeout(const mpp::TaskMeta & meta, std::
...
-        : task(other.task)
-        , reason(other.reason)
-    {
-        LOG_FMT_WARNING(getLogger(), "Copy constructor of MPPTaskCancelFunctor called, should not happen");
-    }
-    MPPTaskCancelFunctor(MPPTaskCancelFunctor && other) = default;
-    MPPTaskCancelFunctor(MPPTaskPtr && task_, const String & reason_)
+    MPPTaskCancelHelper(const MPPTaskCancelHelper & other) = delete;
+    MPPTaskCancelHelper(MPPTaskCancelHelper && other) = delete;
+    MPPTaskCancelHelper(MPPTaskPtr && task_, const String & reason_)
```
Comment (gengliqi) [inline dbms/src/Flash/Mpp/MPPTaskManager.cpp:80]: How about using `DISALLOW_COPY_AND_MOVE`?

Comment (windtalker) [reply dbms/src/Flash/Mpp/MPPTaskManager.cpp]: Updated.

**Review by gengliqi — APPROVED** [review]
LGTM

**Review by fuzhe1989 — APPROVED** [review]

**Pull request merged**